pkg/openstack/loadbalancer.go

```diff
@@ -2317,6 +2364,12 @@ func (lbaas *LbaasV2) ensureAndUpdateOctaviaSecurityGroup(clusterName string, ap
 			return fmt.Errorf("failed to create Security Group for loadbalancer service %s/%s: %v", apiService.Namespace, apiService.Name, err)
 		}
 		lbSecGroupID = lbSecGroup.ID
+
+		tags := getCustomLoadBalancerTags(apiService, svcConf)
+
+		if _, err := neutrontags.ReplaceAll(lbaas.network, "security-groups", lbSecGroupID, neutrontags.ReplaceAllOpts{Tags: tags}).Extract(); err != nil {
+			return fmt.Errorf("failed to add custom tags %s to security group %s (%s)", tags, lbSecGroupID, lbSecGroupName)
+		}
```

Comment:
This is missing the `mc = metrics.NewMetricContext("security_group_tag", "replace")` and then the subsequent `ObserveRequest()`.

Reply:
Same here should only do tagging when `len(tags) > 0`.